src/main/java/com/uber/profiling/Arguments.java

```diff
@@ -227,10 +247,35 @@ public Reporter getReporter() {
         }
     }
 
+    public ConfigProvider getConfigProvider() {
+        if (configProviderConstructor == null) {
+            return new DummyConfigProvider();
+        } else {
+            try {
+                ConfigProvider configProvider = configProviderConstructor.newInstance();
+                if (configProvider instanceof YamlConfigProvider) {
+                    if (configFile == null || configFile.isEmpty()) {
+                        logger.warn("Argument configFile is empty, cannot use " + configProvider.getClass());
+                        return new DummyConfigProvider();
+                    }
```

Comment:
To pursue a cleaner design, I would suggest putting validation logic in the implementation of ConfigProvider instead of here, 

specifically for this case, I think it is fine to impose that configFile and configProviderConstructor should be specified simultaneously, so that we can validate in https://github.com/uber-common/jvm-profiler/pull/1/files#diff-067db4dea723c24224a1fba42569d014R130 avoiding leaking the logic related to a specific config provider implementation to the factory method

Reply:
It is intended to keep argument checking outside of YamlConfigProvider, so YamlConfigProvider is small and simple. Also that gives more flexible, for example, we could switch to DummyConfigProvider if configFile not provided.

Reply:
After in-person discussion, changed the code to throw exception on null or empty configFile here.